numba/targets/arraymath.py

```diff
@@ -999,3 +1019,210 @@ def dot_3(context, builder, sig, args):
+                 lwork, info):
+    """
+    Call the LAPACK gettri function for the given arguemnt.
+
+    This function compute the invers of a matrix given its LU decomposition.
+
+    """
+    fnty = ir.FunctionType(ir.IntType(32),
+                           [ll_char,                       # kind
+                            intp_t, ll_void_p, intp_t,     # n, a, lda
+                            ll_intc_p, ll_void_p,          # ipiv, work
+                            ll_intc_p, ll_intc_p           # lwork, info
+                           ])
+    fn = builder.module.get_or_insert_function(fnty, name="numba_xxgetri")
+
+    kind = get_blas_kind(a_type.dtype)
+    kind_val = ir.Constant(ll_char, ord(kind))
+
+    n = lda = a_shapes[0]
+
+    res = builder.call(fn, (kind_val, n,
+                            builder.bitcast(a_data, ll_void_p), lda,
+                            ipiv, builder.bitcast(work, ll_void_p),
+                            lwork, info
+                            ))
+    check_lapack_return(context, builder, res)
+
+
+def mat_inv(context, builder, sig, args):
+    """
+    Invert a matrix by first computing its LU decomposition.
+    """
+    xty = sig.args[0]
+    dtype = xty.dtype
+
+    x = make_array(xty)(context, builder, args[0])
+    x_shapes = cgutils.unpack_tuple(builder, x.shape)
+    m, n = x_shapes
+    check_c_int(context, builder, m)
+    check_c_int(context, builder, n)
+
+    def create_out(a):
+        m, n= a.shape
+        if m != n:
+            raise ValueError("np.linalg.inv can only work on square "
+                             "arrays.")
+        return a.copy()
+
+    out = context.compile_internal(builder, create_out,
+                                   signature(xty, *sig.args), args)
+    o = make_array(xty)(context, builder, out)
+
+
+    def alloca_ipiv(a):
+        """Allocate the pivot vector.
+
+        """
+        return numpy.empty((a.shape[0],), numpy.int32)
+
+    ipiv_t = types.Array(types.intc, 1, 'C')
+    ipiv = context.compile_internal(builder, alloca_ipiv,
+                                   signature(ipiv_t, *sig.args), args)
+    i = make_array(ipiv_t)(context, builder, ipiv)
+
+    info = cgutils.alloca_once(builder, ll_intc)
+
+    # Compute the LU decomposition of the matrix.
+    call_xxgetrf(context, builder, xty, x_shapes, o.data, i.data,
+                 info)
+
+    zero = ir.Constant(ll_intc, 0)
+    lapack_error = builder.icmp_signed('!=', info, zero)
+    invalid_arg = builder.icmp_signed('<', info, zero)
+
+    def lapack_invalid_arg(i):
+        raise ValueError('%d-th argument pased to getrf is invalid' % i)
+
+    def lapack_singular():
+        raise ValueError('Matrix is singular and cannot be inverted')
+
+    with builder.if_then(lapack_error, False):
+        with builder.if_else(invalid_arg) as (then, otherwise):
+            with then:
+                err_id = builder.load(info)
+                # XXX this raises NotImplementedError $0.5
```

Comment:
Right, this is because string formatting is not supported at runtime. You should simply use a constant error message as a workaround.

(also, you don't strictly need compile_internal(), instead you can use something such as:
```
            context.call_conv.return_user_exc(builder, ValueError,
                                              ("Matrix is singular and cannot be inverted",))
```
but that needn't block you from progressing)